examples/vector_fe/vector_fe_ex7/vector_fe_ex7.C

```diff
@@ -0,0 +1,551 @@
+// The libMesh Finite Element Library.
+// Copyright (C) 2002-2023 Benjamin S. Kirk, John W. Peterson, Roy H. Stogner
+
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+// Basic utilities.
+#include "libmesh/string_to_enum.h"
+
+// The solver packages supported by libMesh.
+#include "libmesh/enum_solver_package.h"
+
+// The mesh object and mesh generation and modification utilities.
+#include "libmesh/mesh.h"
+#include "libmesh/mesh_generation.h"
+#include "libmesh/mesh_modification.h"
+
+// Matrix and vector types.
+#include "libmesh/dense_matrix.h"
+#include "libmesh/sparse_matrix.h"
+#include "libmesh/dense_vector.h"
+#include "libmesh/numeric_vector.h"
+
+// The finite element object and the geometric element type.
+#include "libmesh/fe.h"
+#include "libmesh/elem.h"
+
+// Gauss quadrature rules.
+#include "libmesh/quadrature_gauss.h"
+
+// The dof map, which handles degree of freedom indexing.
+#include "libmesh/dof_map.h"
+
+// The system of equations.
+#include "libmesh/equation_systems.h"
+#include "libmesh/linear_implicit_system.h"
+
+// The exact solution and error computation.
+#include "libmesh/exact_solution.h"
+#include "libmesh/enum_norm_type.h"
+#include "solution_function.h"
+
+// I/O utilities.
+#include "libmesh/getpot.h"
+#include "libmesh/exodusII_io.h"
+
+#include <Eigen/Dense>
```

Comment:
Let's get an `#ifdef LIBMESH_HAVE_EIGEN_DENSE` here somewhere?  I know everybody configures it on but we're at least still trying to keep it as an optional requirement.